### ISSUE TITLE
[amp-stories] Sharing is broken on mobile devices

### ISSUE THREAD

**Issue by cpauwels** [ampproject/amphtml#27826]
## What's the issue?

On both Android and iOS the share button in the top right does absolutely nothing.
 
## How do we reproduce the issue?

Open any AMP story from your mobile device and tap the share icon.

## What browsers are affected?

Tested on Chrome (Android) and Safari (iOS)

## Which AMP version is affected?

Probably the latest release.

**Comment by newmuis** [comment]
@Enriqe @gmajoulet can you take a look and see what's causing this?

**Comment by Enriqe** [comment]
I'm ablo to repro in version AMP version `2004161906340` on iPhone.

```
[Error] Error: Clicked before href is set.
	(anonymous function) (v0.js:25:416)
	C (amp-social-share-0.1.js:2:1373)
	W (amp-social-share-0.1.js:16)
	(anonymous function) (amp-social-share-0.1.js:13:317)
	dispatchEvent
	Pj (amp-story-1.0.js:338)
	(anonymous function) (amp-story-1.0.js:337:305)
	(anonymous function) (amp-story-1.0.js:25:741)
	(anonymous function) (amp-story-1.0.js:40:199)
	forEach
	(anonymous function) (amp-story-1.0.js:40:132)
	(anonymous function) (amp-story-1.0.js:362:419)
```

**Comment by Enriqe** [comment]
Looks like #27533 moved the assignment of `href` from being set in the `buildCallback()` to being set in `layouCallback()` in `AmpSocialShare`. But for some reason `layoutCallback()` isn't being called in `AmpSocialShare`. Investigating...

**Comment by Enriqe** [comment]
Problem is we append a hidden `amp-story-share` component here
https://github.com/ampproject/amphtml/blob/7d820920d6c9d890b0f15e6d69a721dfde92ea9e/extensions/amp-story/1.0/amp-story-share-menu.js#L137

https://github.com/ampproject/amphtml/blob/7d820920d6c9d890b0f15e6d69a721dfde92ea9e/extensions/amp-story/1.0/amp-story-share-menu.js#L148

so `AmpSocialShare.layoutCallback()` never gets called.

**Comment by newmuis** [comment]
Do we have a sense of solutions here?

Since it's late on a Friday with an already-crazy release, @caroqliu do you know if #27533 would be a clean rollback?  We could then patch the rollback and then roll forward with a fix.

**Comment by Enriqe** [comment]
I sent a pr with a fix in #27832

**Comment by caroqliu** [comment]
Yes, #27533 would be a clean rollback.
